app/models/topic.rb

```diff
@@ -131,4 +133,12 @@ def topic_autocomplete_method
   def mk_permission
     Perm.short(permission)
   end
+
+  protected
+    def create_metamap?
+      if link == '' and metacode.name == 'Metamap'
+        @map = Map.create({ name: name, permission: permission, desc: '', arranged: true, user_id: user_id })
+        self.link = Rails.application.routes.url_helpers.map_url(:host => ENV['MAILER_DEFAULT_URL'], :id => @map.id)
```

Comment:
you mean an environment variable? I don't know if base_ur would work, as this code is in a Model